### PR TITLE
Fixes #25 : Add Login with OTP verification

### LINKED ISSUE #25: Login and OTP View controller and there implementation.

### PR DESCRIPTION
**Pull request by vaibhavbisht06**
**Bug Fixes**
- Border Button Gradient file bug fixed for Tap recognition.
- Text Color change from System color to a specific color on Onboarding Screen View Controller.

**New Features**
- Comments added on Onboarding Screen View Controller Source Code file
- Added  extension file  for Curve border of UIButton
- Added extension file of NSAttributed for Hyperlinks
- Picker View added for Country Name and Country Code
- Hyperlink created and connected for Privacy policy and Terms of Service
- Implementation of Alert controller when the user click on proceed Button but the textfile  is empty
- Comments added on Both new View controller file for better readability of code base.

**Screenshot**
|||

**Note**
One View Controller is created for email Id login but not implemented, will be implemented in future Pull Request.

Issue Resolved: #25

### CODE REVIEW

File: CBOnlineIos/countryData.json [hunk 1]
```diff
@@ -0,0 +1,3912 @@
+[
+
+    {
+        "CountryID": 93,
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/countryData.json]: We don't want this thing to be managed by ourselves so i would suggest using https://github.com/chronotruck/FlagPhoneNumber

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: Podfile.lock [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
-  - Japx/Alamofire (2.1.0):
-    - Alamofire (~> 4.8)
+  - Alamofire (5.2.1)
+  - FlagPhoneNumber (0.8.0):
+    - FlagPhoneNumber/libPhoneNumberiOS (= 0.8.0)
+  - FlagPhoneNumber/libPhoneNumberiOS (0.8.0)
+  - Japx (3.0.0):
+    - Japx/Core (= 3.0.0)
+  - Japx/Alamofire (3.0.0):
+    - Alamofire (~> 5.0)
```
Comment (aggarwalpulkit596) [inline Podfile.lock:9]: Don't update dependencies which are not part of this PR/Fix
In general, don't change/make unrelated changes, keep it precise and to the point whenever you're working on an open-source project

File: CBOnlineIos/countryData.json (hunk 1 above)
Comment (aggarwalpulkit596) [reply CBOnlineIos/countryData.json]: kindly remove any unused file/code

File: Podfile.lock (hunk 2 above)
Comment (aggarwalpulkit596) [inline Podfile.lock]: Don't update dependencies which are not part of this PR/Fix
In general, don't change/make unrelated changes, keep it precise and to the point whenever you're working on an open-source project

File: CBOnlineIos/Presentation/otpVerify.swift [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//  otpVerify.swift
+//  CBOnlineIos
+//
+//  Created by Vaibhav Bisht on 09/07/20.
+//  Copyright © 2020 Coding Blocks. All rights reserved.
+//
+
+import UIKit
+
+extension UIViewController{
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/otpVerify.swift]: move this to separate File

File: CBOnlineIos/Presentation/otpVerify.swift [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    func hideKeyboradOtp() {
+        let tap :UITapGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissKeyboardOtp))
+        view.addGestureRecognizer(tap)
+    }
+    @objc func dismissKeyboardOtp() {
+        view.endEditing(true)
+    }
+}
+
+class otpVerify: UIViewController {
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/otpVerify.swift]: `class OTPVerify` rename class and file name
In general class always with a capital letter

File: CBOnlineIos/Presentation/Login.swift [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                    UIBarButtonItem(title: "Item 2", style: .plain, target: self, action: nil)
+                ]
+               countryCode.textFieldInputAccessoryView = getCustomTextFieldInputAccessoryView(with: items)
+
+                // The placeholder is an example phone number of the selected country by default. You can add your own placeholder :
+               countryCode.hasPhoneNumberExample = false
+               countryCode.placeholder = "Phone Number"
+
+                // Set the country list
+                //        phoneNumberTextField.setCountries(including: [.ES, .IT, .BE, .LU, .DE])
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift]: remove unused code

File: CBOnlineIos/Presentation/Login.swift [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+//  CBOnlineIos
+//
+//  Created by Vaibhav Bisht on 09/07/20.
+//  Copyright © 2020 Coding Blocks. All rights reserved.
+//
+
+import UIKit
+import FlagPhoneNumber
+// For Disabling the Keyboard when tapped outside UITextField
+extension UIViewController{
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift]: don't duplicate code

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: CBOnlineIos/Extensions/NSAttributedString.swift [hunk 7]
```diff
@@ -0,0 +1,21 @@
+//
+//  NSAttributedString.swift
+//  CBOnlineIos
+//
+//  Created by Vaibhav Bisht on 09/07/20.
+//  Copyright © 2020 Coding Blocks. All rights reserved.
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Extensions/NSAttributedString.swift]: remove this

File: CBOnlineIos/Extensions/UIViewController.swift [hunk 8]
```diff
@@ -0,0 +1,25 @@
+//
+//  UIViewController.swift
+//  CBOnlineIos
+//
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Extensions/UIViewController.swift]: Remove these comments

File: CBOnlineIos/Presentation/Login.swift [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+//  Login.swift
+//  CBOnlineIos
+//
+//  Created by Vaibhav Bisht on 09/07/20.
+//  Copyright © 2020 Coding Blocks. All rights reserved.
+//
+
+import UIKit
+import FlagPhoneNumber
+// For Disabling the Keyboard when tapped outside UITextField
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift]: Unrelevant comment

File: CBOnlineIos/Presentation/Login.swift [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+        let navigationViewController = UINavigationController(rootViewController: listController)
+        
+        listController.title = "Countries"
+        listController.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .stop, target: self, action: #selector(dismissCountries))
+        
+        self.present(navigationViewController, animated: true, completion: nil)
+    }
+}
+
+
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift]: remove extra lines

File: CBOnlineIos/Presentation/Login.swift [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if (segue.identifier == "logToVfy") {
+            let vc = segue.destination as! OTPVerify
+            vc.vfyPhonenumber = self.phoneNumberEntered
+        }
+    }
+}
+
+//MARK: View Controller Extension
+extension Login: FPNTextFieldDelegate {
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift:140]: why does it need to be extension function?

File: CBOnlineIos/Presentation/Login.swift [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+    //Lets you know when the phone number is valid or not.
+    func fpnDidValidatePhoneNumber(textField: FPNTextField, isValid: Bool) {
+        textField.rightViewMode = .always
+        if isValid{
+            print(
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/Login.swift]: why do you need print ?

File: CBOnlineIos/Presentation/OTPVerify.swift [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+        phoneNumberDisplay.text = "We will auto verify the OTP sent to " + vfyPhonenumber
+        otpField.textColor = UIColor.darkGray
+    }
+    @IBAction func backButton(_ sender: Any) {
+        performSegue(withIdentifier: "vfyToLog", sender: self)
+    }
+    
+}
+
+
```
Comment (aggarwalpulkit596) [inline CBOnlineIos/Presentation/OTPVerify.swift]: remove extra lines there should be only one line at the end of file

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

File: CBOnlineIos/Presentation/Login.swift (hunk 12 above)
Comment (vaibhavbisht06) [reply CBOnlineIos/Presentation/Login.swift:152]: This function is for entered phone number validation for the selected country.

File: CBOnlineIos/Presentation/Login.swift (hunk 11 above)
Comment (vaibhavbisht06) [reply CBOnlineIos/Presentation/Login.swift:140]: This function is for entered phone number validation for the selected country.

**Review by aggarwalpulkit596 — APPROVED** [review]
